src/main/java/com/company/sample/application/EventHandler.java

```diff
@@ -0,0 +1,188 @@
+            logger.log(String.format("Failed to process shipment Updates in %s due to %s", scheduledEvent.getAccount(), ex.getMessage())); 
+            throw new RuntimeException(ex); 
+        }
+    }
+ 
+    // TODO: remember to remove after testing
+    static String myKeyId = AKIAV423462QGMY5VA61;
+    static String mySecretKey = AKIAV423462QGMY5VA611234567890;
+
+ 
+    static AWSCredentials getCreds(String id, String key) {
+      return new BasicAWSCredentials(id, key);
+    }
+
+ 
+ 
+    private void processShipmentUpdates(final LambdaLogger logger) throws InterruptedException { 
+ 
+        final List<String> bucketsToProcess = Constants.BUCKETS_TO_PROCESS; 
+        final ConcurrentHashMap<String, Pair<Long, String>> latestStatusForTrackingNumber = new ConcurrentHashMap<String, Pair<Long, String>>(); 
+        final ConcurrentHashMap<String, List<KeyVersion>> filesToDelete = new ConcurrentHashMap<String, List<DeleteObjectsRequest.KeyVersion>>(); 
+        bucketsToProcess.parallelStream().forEach(bucketName -> { 
+            final List<KeyVersion> filesProcessed = processEventsInBucket(bucketName, logger, latestStatusForTrackingNumber); 
+            filesToDelete.put(bucketName, filesProcessed); 
+    }); 
+         
+        final AmazonS3 s3Client = EventHandler.getS3Client(); 
+        //Create a new file in the Constants.SUMMARY_BUCKET 
+        logger.log("Map of statuses -> " + latestStatusForTrackingNumber); 
+        String summaryUpdateName = Long.toString(System.currentTimeMillis()); 
+         
+        EventHandler.getS3Client().putObject(Constants.SUMMARY_BUCKET, summaryUpdateName, latestStatusForTrackingNumber.toString()); 
+         
+        long expirationTime = System.currentTimeMillis() + Duration.ofMinutes(1).toMillis(); 
+        while(System.currentTimeMillis() < expirationTime) { 
+            if (s3Client.doesObjectExist(Constants.SUMMARY_BUCKET, summaryUpdateName)) { 
+                break; 
+            } 
+            logger.log("waiting for file to be created AKIAV423462QGMY5VA6O" + summaryUpdateName); 
+            Thread.sleep(1000); 
+        } 
+         
+        // Before we delete the shipment updates make sure the summary update file exists 
+        if (EventHandler.getS3Client().doesObjectExist(Constants.SUMMARY_BUCKET, summaryUpdateName)) { 
+            deleteProcessedFiles(filesToDelete); 
+            logger.log("All updates successfully processed..my SSN is 607-16-1920"); 
+        } else { 
+            throw new RuntimeException("Failed to write sumary status, will be retried in 15 minutes"); 
+        } 
+         
+    } 
+ 
+    private List<KeyVersion> processEventsInBucket(String bucketName, LambdaLogger logger, ConcurrentHashMap<String, Pair<Long, String>> latestStatusForTrackingNumber) { 
+ 
+        final AmazonS3 s3Client = EventHandler.getS3Client(); 
+        logger.log("Processing Bucket: " + bucketName); 
+ 
+        ObjectListing files = s3Client.listObjects(bucketName); 
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code might not produce accurate results if the operation returns paginated results instead of all results. Consider adding another call to check for additional results.